### PR TITLE
Feature 18

### LINKED ISSUE #18: An update activity endpoint
**Is your feature request related to a problem? Please describe.**
We would like a new endpoint capable of updating a given activity. Currently we are required to send the entire weekmodel to the API to update even the smallest things, but with this new endpoint we can minimize the data sent to the API.

**Describe the solution you'd like**
We want an endpoint which only takes an activity and updates it in the database. The request would need metadata attached to it as the API needs to know which user and weekmodel the activity belongs to.

**Describe alternatives you've considered**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by HaZzM22**
Closes #18 

Merged from #17

### CODE REVIEW

**Comment by KVISDAOWNER** [comment]
@aau-giraf/sw609f19  you have been assigned this. Here's your checklist. I marked @Last16  as the reviewer, but you might change it as you like.
Please use the following checklist. Feel free to check it off as you go along.
**Code Design**
- [ ] Is the code in the right place? (Both in terms of folder structure and class structure)
- [ ] Could this code, to the best of your knowledge, have reused existing code?
- [ ] Is the code over-engineered? Examples of over-engineering: Implemented behavior for future problems
- [ ] Does it introduce functionality that is not necessary for solving the problem?

**Code Readability and Maintainability**
- [ ] Are names meaningful and self-documenting?
- [ ] Is it understandable, by reading the code, what it does?
- [ ] Is the code simple and readable? i.e. it contains no “hack”, or obscure solution.
- [ ] Is it understandable, by reading the tests, what they do?
- [ ] Do the tests cover sensible cases? Both happy paths and exception paths?
- [ ] Do the tests cover the full functionality? i.e., will the tests fail if some of the requested functionality is missing?
- [ ] Are error messages in the code understandable?
- [ ] Are confusing sections of code covered by comments?

**Functionality**
For this part, open the app, and test:
- [ ] Are the features that the code claims to implement, actually implemented?
- [ ] Does it look like the code has bugs?
- [ ] Are all new screens reachable through in-app navigation?

**Points for Discussion and Reflection**
- [ ] Are UI messages, that the user might meet, actually helpful?
- [ ] Are there errors that are bad practice? Like, hard-coding a test database connection?
- [ ] Is there well written code in this? Remember to tell the developer!

**Comment by KVISDAOWNER** [comment]
@aau-giraf/sw602f19 you have been assigned this. Here's your checklist. I marked @KVISDAOWNER as the reviewer, but you might change it as you like.
Please use the following checklist. Feel free to check it off as you go along.
**Code Design**
- [ ] Is the code in the right place? (Both in terms of folder structure and class structure)
- [ ] Could this code, to the best of your knowledge, have reused existing code?
- [ ] Is the code over-engineered? Examples of over-engineering: Implemented behavior for future problems
- [ ] Does it introduce functionality that is not necessary for solving the problem?

**Code Readability and Maintainability**
- [ ] Are names meaningful and self-documenting?
- [ ] Is it understandable, by reading the code, what it does?
- [ ] Is the code simple and readable? i.e. it contains no “hack”, or obscure solution.
- [ ] Is it understandable, by reading the tests, what they do?
- [ ] Do the tests cover sensible cases? Both happy paths and exception paths?
- [ ] Do the tests cover the full functionality? i.e., will the tests fail if some of the requested functionality is missing?
- [ ] Are error messages in the code understandable?
- [ ] Are confusing sections of code covered by comments?

**Functionality**
For this part, open the app, and test:
- [ ] Are the features that the code claims to implement, actually implemented?
- [ ] Does it look like the code has bugs?
- [ ] Are all new screens reachable through in-app navigation?

**Points for Discussion and Reflection**
- [ ] Are UI messages, that the user might meet, actually helpful?
- [ ] Are there errors that are bad practice? Like, hard-coding a test database connection?
- [ ] Is there well written code in this? Remember to tell the developer!

**Comment by Tricky12321** [comment]
You need to merge into develop and not master!

File: GirafRest.Test/Controllers/ActivityControllerTest.cs [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+            _testContext.MockUserManager.MockLoginAsUser(mockUser);
+            Week week = mockUser.WeekSchedule.First();
+
+            ActivityDTO newActivity = new ActivityDTO() {Id=420};
+
+            var res = ac.UpdateActivity(newActivity).Result;
+
+            Assert.False(res.Success);
+            Assert.Equal(ErrorCode.ActivityNotFound, res.ErrorCode);
+        }
```
Comment (KVISDAOWNER) [inline GirafRest.Test/Controllers/ActivityControllerTest.cs]: Missing the test case where request is succesfull

File: GirafRest/Controllers/PictogramController.cs [hunk 2]
```diff
@@ -131,7 +117,7 @@ public async Task<Response<WeekPictogramDTO>> CreatePictogram([FromBody]Pictogra
             var user = await _giraf.LoadUserWithResources(HttpContext.User);
 
             if (user == null) 
-                return new ErrorResponse<WeekPictogramDTO>(ErrorCode.NotFound);
+                return new ErrorResponse<WeekPictogramDTO>(ErrorCode.ActivityNotFound);
```
Comment (KVISDAOWNER) [inline GirafRest/Controllers/PictogramController.cs]: Is this the correct errorcode, ActivityNotFound? I would expect a UserNotFound

File: GirafRest/Controllers/PictogramController.cs [hunk 3]
```diff
@@ -184,7 +170,7 @@ public async Task<Response<WeekPictogramDTO>> UpdatePictogramInfo(long id, [From
                 return new ErrorResponse<WeekPictogramDTO>(ErrorCode.InvalidProperties);
 
             var usr = await _giraf.LoadBasicUserDataAsync(HttpContext.User);
-            if (usr == null) return new ErrorResponse<WeekPictogramDTO>(ErrorCode.NotFound);
+            if (usr == null) return new ErrorResponse<WeekPictogramDTO>(ErrorCode.ActivityNotFound);
```
Comment (KVISDAOWNER) [inline GirafRest/Controllers/PictogramController.cs]: Is this the correct errorcode, ActivityNotFound? I would expect a UserNotFound

**Comment by KVISDAOWNER** [comment]
You could also consider if a PATCH request would be a more suitable request method.

**Pull request closed without merging**